client.go

```diff
@@ -74,13 +84,15 @@ func New(c Config) Client {
 		c.RetryConfig,
 	)
 
-	return NewClient(httpClient)
+	return NewClient(httpClient, GetApiTld(&c))
 }
 
 // NewClient creates a new Site24x7 API Client from httpClient. This can be
 // used to provide a custom http client for use with the API. The custom http
 // client has to transparently handle the Site24x7 OAuth flow.
-func NewClient(httpClient HTTPClient) Client {
+func NewClient(httpClient HTTPClient, apiTld string) Client {
```

Comment:
This is a breaking change. I suggest to add `NewClientWithBaseURL(*http.Client, string)` and call this from `New` instead if the `APIBaseURL` is not empty.

Reply:
Changed in https://github.com/Bonial-International-GmbH/site24x7-go/pull/34